### PR TITLE
[form recognizer] Remove US receipt

### LINKED ISSUE #11712: [form recognizer] remove USReceipt
See .net issue [here](https://github.com/Azure/azure-sdk-for-net/issues/11912)

We will remove USReceipt class and related types

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #11712

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,8 @@
 `CustomFormModel`  and `CustomFormModelInfo` models.
 - `models` property of `CustomFormModel` is renamed to `submodels`
 - `CustomFormSubModel` is renamed to `CustomFormSubmodel`
+- Removed `USReceipt`. To see how to deal with the return value of `begin_recognize_receipts`, see [sample_recognize_receipts.py](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/samples/sample_recognize_receipts.py) or [sample_recognize_receipts_from_url.py](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/samples/sample_recognize_receipts_from_url.py) for details.
+- Removed `USReceiptItem`. To see how to access the individual items on a receipt, see [sample_recognize_receipts.py](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/samples/sample_recognize_receipts.py) or [sample_recognize_receipts_from_url.py](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/formrecognizer/azure-ai-formrecognizer/samples/sample_recognize_receipts_from_url.py) for details.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:26]: Should we just link to the samples directory? Part of me worries that we might rename these samples one day and not think to check the Changelog for needing update.

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 2]
```diff
@@ -227,21 +227,40 @@ with open("<path to your receipt>", "rb") as fd:
 poller = form_recognizer_client.begin_recognize_receipts(receipt)
 result = poller.result()
 
-r = result[0]
-print("Receipt contained the following values with confidences: ")
-print("Receipt Type: {} has confidence: {}".format(r.receipt_type.type, r.receipt_type.confidence))
-print("Merchant Name: {} has confidence: {}".format(r.merchant_name.value, r.merchant_name.confidence))
-print("Transaction Date: {} has confidence: {}".format(r.transaction_date.value, r.transaction_date.confidence))
+receipt = result[0]
+print("Receipt Type: {} has confidence: {}".format(receipt.receipt_type.type, receipt.receipt_type.confidence))
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:231]: should `receipt_type` be strongly typed?

edit: asking Anne for clarity on this

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_receipts_async.py [hunk 3]
```diff
@@ -46,18 +46,38 @@ async def recognize_receipts(self):
...
+                    print("Subtotal: {} has confidence: {}".format(subtotal.value, subtotal.confidence))
+                tax = receipt.fields.get("Tax")
+                if tax:
+                    print("Tax: {} has confidence: {}".format(tax.value, tax.confidence))
+                tip = receipt.fields.get("Tip")
+                if tip:
+                    print("Tip: {} has confidence: {}".format(tip.value, tip.confidence))
+                total = receipt.fields.get("Total")
+                if total:
+                    print("Total: {} has confidence: {}".format(total.value, total.confidence))
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_receipts_async.py:80]: I understand wanting to show the keys available, but wondering if we should do at least one sample or readme like this:

```
for receipt in result:
    for key, val in receipt.fields.items():
        print("{}: {} has confidence: {}".format(key, val.value, val.confidence))
```

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 2 above)
Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md]: After guidance from @annelo-msft, we will treat this as any other field

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_receipts_async.py (hunk 3 above)
Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_receipts_async.py:82]: I see what you mean. I think I will leave the samples like this, but update the readme sample to traverse the fields like this

Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_receipts_async.py:83]: I really like how the readme looks. 
For this sample, I will suggest adding a comment on top a [link ](https://westus2.dev.cognitive.microsoft.com/docs/services/form-recognizer-api-v2-preview/operations/GetAnalyzeReceiptResult) to the types the service is currently supporting, or guiding the user into finding this information

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md [hunk 4]
```diff
@@ -227,21 +227,16 @@ with open("<path to your receipt>", "rb") as fd:
...
-    print("...Total Item Price: {} has confidence: {}".format(item.total_price.value, item.total_price.confidence))
-print("Subtotal: {} has confidence: {}".format(r.subtotal.value, r.subtotal.confidence))
-print("Tax: {} has confidence: {}".format(r.tax.value, r.tax.confidence))
-print("Tip: {} has confidence: {}".format(r.tip.value, r.tip.confidence))
-print("Total: {} has confidence: {}".format(r.total.value, r.total.confidence))
+for receipt in result:
+    for name, field in receipt.fields.items():
+        if name == "Items":
+            print("Receipt Items:")
+            for idx, items in enumerate(field.value):
```
Comment (annelo-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/README.md:234]: If it's at all complicated to retrieve these values, it's probably worth illustrating for customers how to get them out of the receipt fields property.

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_receipts_async.py (hunk 3 above)
Comment (annelo-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_receipts_async.py:83]: For the .NET sample, we're only showing a subset of illustrative values from the receipt.

Also, I'm not sure how it works in Python, but do you need to check the FieldValueType  to cast these to the right time in order for a customer to use them in a scenario?

File: sdk/formrecognizer/azure-ai-formrecognizer/README.md (hunk 4 above)
Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/README.md:234]: We are still accessing the field properties (including items) one by one in the [samples](https://github.com/Azure/azure-sdk-for-python/blob/985a89432412713787817a66f86c55f176f6291d/sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_recognize_receipts_async.py#L39). For the readme, based on @kristapratico , I wanted to show a simpler way of traversing everything

**Pull request merged**